dot/rpc/modules/offchain_test.go

```diff
@@ -1 +1,210 @@
+				},
+			},
+			expErr: errors.New("cannot decode an odd length string"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &OffchainModule{
+				nodeStorage: tt.fields.nodeStorage,
+			}
+			res := StringResponse("")
+			err := s.LocalStorageGet(tt.args.in0, tt.args.req, &res)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.exp, res)
+		})
+	}
+}
+
+func TestOffchainModule_LocalStorageSet(t *testing.T) {
+	mockRuntimeStorageAPI := new(apimocks.RuntimeStorageAPI)
+	mockRuntimeStorageAPI.On("SetLocal", common.MustHexToBytes("0x11111111111111"), common.MustHexToBytes("0x22222222222222")).Return(nil)
+	mockRuntimeStorageAPI.On("SetPersistent", common.MustHexToBytes("0x11111111111111"), common.MustHexToBytes("0x22222222222222")).Return(errors.New("SetPersistent error"))
+
+	type fields struct {
+		nodeStorage RuntimeStorageAPI
+	}
+	type args struct {
+		in0 *http.Request
+		req *OffchainLocalStorageSet
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		expErr error
+	}{
+		{
+			name: "setLocal OK",
+			fields: fields{
+				mockRuntimeStorageAPI,
+			},
+			args: args{
+				req: &OffchainLocalStorageSet{
+					Kind:  offchainLocal,
+					Key:   "0x11111111111111",
+					Value: "0x22222222222222",
+				},
+			},
+		},
+		{
+			name: "Invalid Key",
+			fields: fields{
+				mockRuntimeStorageAPI,
+			},
+			args: args{
+				req: &OffchainLocalStorageSet{
+					Kind:  offchainLocal,
+					Key:   "0x1",
+					Value: "0x22222222222222",
+				},
+			},
+			expErr: errors.New("cannot decode an odd length string"),
+		},
+		{
+			name: "Invalid Value",
+			fields: fields{
+				mockRuntimeStorageAPI,
+			},
+			args: args{
+				req: &OffchainLocalStorageSet{
+					Kind:  offchainLocal,
+					Key:   "0x11111111111111",
+					Value: "0x2",
+				},
+			},
+			expErr: errors.New("cannot decode an odd length string"),
+		},
+		{
+			name: "setPersistentError",
+			fields: fields{
+				mockRuntimeStorageAPI,
+			},
+			args: args{
+				req: &OffchainLocalStorageSet{
+					Kind:  offchainPersistent,
+					Key:   "0x11111111111111",
+					Value: "0x22222222222222",
+				},
+			},
+			expErr: errors.New("SetPersistent error"),
+		},
+		{
+			name: "Invalid Kind",
+			fields: fields{
+				mockRuntimeStorageAPI,
+			},
+			args: args{
+				req: &OffchainLocalStorageSet{
+					Kind:  "bad kind",
+					Key:   "0x11111111111111",
+					Value: "0x22222222222222",
+				},
+			},
+			expErr: fmt.Errorf("storage kind not found: bad kind"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &OffchainModule{
+				nodeStorage: tt.fields.nodeStorage,
+			}
+			res := StringResponse("")
+			err := s.LocalStorageSet(tt.args.in0, tt.args.req, &res)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
```

Comment:
assert the response perhaps?

Reply:
there is no response set in this function

Reply:
Isn't there that `res := StringResponse("")`? Since it's passed as pointer, even if it's not modified, you can assert it's untouched too.